dev/WebView2/WebView2.cpp

```diff
@@ -0,0 +1,1942 @@
+
+    if (!m_options)
+    {
+        // NOTE: To enable Anaheim logging, use: L"--no-sandbox --enable-features=msEmbeddedBrowserVisualHosting --enable-logging=stderr --v=1"
+        m_options = winrt::CoreWebView2EnvironmentOptions();
+        m_options.AdditionalBrowserArguments(L"--enable-features=msEmbeddedBrowserVisualHosting");
+
+        auto applicationLanguagesList = winrt::ApplicationLanguages::Languages();
+        if (applicationLanguagesList.Size() > 0)
+        {
+            m_options.Language(applicationLanguagesList.GetAt(0));
+        }
+    }
+
+    try
+    {
+        m_coreWebViewEnvironment = co_await winrt::CoreWebView2Environment::CreateWithOptionsAsync(browserInstall, userDataFolder, m_options);
+    }
+    catch (winrt::hresult_error e)
+    {
+        winrt::hresult hr = e.code().value;
+        m_shouldShowMissingAnaheimWarning = hr == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
+        FireCoreWebView2Initialized(hr);
+    }
+
+    co_return;
+}
+
+winrt::IAsyncAction WebView2::CreateCoreWebViewFromEnvironment(HWND hwndParent)
+{
+    if (!hwndParent)
+    {
+        hwndParent = EnsureTemporaryHostHwnd();
+    }
+
+    try
+    {
+        auto windowRef = winrt::CoreWebView2ControllerWindowReference::CreateFromWindowHandle(reinterpret_cast<UINT64>(hwndParent));
+        m_coreWebViewCompositionController = co_await m_coreWebViewEnvironment.CreateCoreWebView2CompositionControllerAsync(windowRef);
+        m_coreWebViewController = m_coreWebViewCompositionController.as<winrt::CoreWebView2Controller>();
+        m_coreWebViewController.ShouldDetectMonitorScaleChanges(false);
+        m_coreWebView = m_coreWebViewController.CoreWebView2();
+        m_everHadCoreWebView = true;
+
+        RegisterCoreEventHandlers();
+        // Creation is considered complete at this point, however rendering and accessibility
+        // hookup will only run after we get Loaded event (see TryCompleteInitialization())
+        FireCoreWebView2Initialized(winrt::hresult(S_OK));
+    }
+    catch (winrt::hresult_error e)
+    {
+        winrt::hresult hr = e.code().value;
+        FireCoreWebView2Initialized(hr);
+    }
+
+    m_isImplicitCreationInProgress = false;
+    m_isExplicitCreationInProgress = false;
+    m_creationInProgressAsync->StopWaiting();
+    m_creationInProgressAsync.reset();
+
+    co_return;
+}
+
+void WebView2::UpdateDefaultBackgroundColor()
+{
+    auto appResources = winrt::Application::Current().Resources();
+    winrt::IInspectable backgroundColorAsI = (m_accessibilitySettings && m_accessibilitySettings.HighContrast()) ?
+        appResources.TryLookup(box_value(L"SystemColorWindowColor")) :
+        appResources.TryLookup(box_value(L"SolidBackgroundFillColorBase"));
+
+    winrt::Color backgroundColor = winrt::unbox_value_or<winrt::Color>(backgroundColorAsI, winrt::Color{});
+#ifdef WINUI3
+    if (m_systemVisualBridge) { m_systemVisualBridge.BackgroundColor(backgroundColor); }
+#endif
```

Comment:
This is needed to prevent unsightly flicker navigating to 1st URI (while CWV2 rendering is being set up) with the SystemVisualBridge design. The BG color should reflect the current theme to avoiad the flicker.

Is the flicker problem not happening with CoreWindow / WUC Visual hosting? If it is, we should add a TODO or task to address it [not sure what would be involved in a fix of that...]

Reply:
We don't use the bridge and i didn't notice any flicker so ... i think we're ok?

Reply:
The flicker is timing-, network-, payload-, and theme-dependent, but it's a good sign you aren't seeing it :-). Agree we can defer this concern until the problem is observed in practice.